Lannister/Data/Repositories Implementations/WalletRepositoryImpl.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  WalletRepositoryImpl.swift
+//  Lannister
+//
+//  Created by Andre Sousa on 12/07/2019.
+//  Copyright © 2019 André Sousa. All rights reserved.
+//
+
+import UIKit
+import MagicalRecord
+import CoreData
+import web3swift
+
+class WalletRepositoryImpl: WalletRepository {
+    
+    func getBalance(address: String, success: @escaping(Double) -> Void, failure: @escaping(_ error: Error) -> Void) {
+        
+        print("getBalance address \(address)")
+        DispatchQueue.global(qos: .background).async {
+            let walletAddress = EthereumAddress(address, ignoreChecksum: true)!
+            let web3 = Web3.InfuraMainnetWeb3(accessToken: "c7b6351e2ba84d3e94d1b33c14bb9a16") // Mainnet Infura Endpoint Provider
+            let balanceResult = try! web3.eth.getBalance(address: walletAddress)
+            let balanceString = Web3.Utils.formatToEthereumUnits(balanceResult, toUnits: .eth, decimals: 3)!
+
+            print("balanceString \(balanceString)")
+            success(balanceString.doubleValue!)
+        }
+    }
+    
+    func getTransactions(address: String, success: @escaping(_ transactions: Array<Transaction>?) -> Void, failure: @escaping(_ error: Error) -> Void) {
+        
+        // module=account&action=txlist&address=0xddbd2b932c763ba5b1b7ae3b362eac3e8d40121a&startblock=0&endblock=99999999&sort=asc&apikey=YourApiKeyToken
+
+        let params = ["module": "account",
+                      "action": "txlist",
+                      "address": address,
+                      "startblock": 0,
+                      "block": 99999999,
+                      "sort": "asc",
+                      "apikey": "GP7SEQH5PPAX47AGETTCU3ZHCMD3WRUP3F"] as [String : Any]
```

Comment:
Maybe make the `apiKey` a variable for the build instead of hard coded? At least refactor to a single file with all the API keys we're using.

Reply:
I'll make it a constant on the respective API service class.